src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,462 @@
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_rvs_1_${t1[0]}$${k1}$     ! 1 dummy variable
+        #:endfor
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_rvs_${t1[0]}$${k1}$       ! 2 dummy variables
+        #:endfor
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_rvs_array_${t1[0]}$${k1}$ ! 3 dummy variables
+        #:endfor
+    end interface uniform_distribution_rvs
+
+    interface uniform_distribution_pdf
+    !! Version experiment
+    !!
+    !! Get uniform distribution probability density (pdf) for integer, real and complex variables
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_pdf_${t1[0]}$${k1}$
+        #:endfor
+    end interface uniform_distribution_pdf
+
+    interface uniform_distribution_cdf
+    !! Version experimental
+    !!
+    !! Get uniform distribution cumulative distribution function (cdf) for integer, real and complex variables
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+    !!
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_cdf_${t1[0]}$${k1}$
+        #:endfor
+    end interface uniform_distribution_cdf
+
+    interface shuffle
+    !! Version experimental
+    !!
+    !! Fisher-Yates shuffle algorithm for a rank one array of integer, real and complex variables
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+    !!
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure shuffle_${t1[0]}$${k1}$
+        #:endfor
+    end interface shuffle
+
+
+    contains
+
+    #:for k1, t1 in INT_KINDS_TYPES
+    impure elemental function unif_dist_rvs_1_${t1[0]}$${k1}$(scale) result(res)
+    ! Uniformly distributed integer in [0, scale]
+    ! Bitmask with rejection
+    ! https://www.pcg-random.org/posts/bounded-rands.html
+    !
+    ! Fortran 90 translated from c by Jim-215-fisher
+        ${t1}$, intent(in) :: scale
+        ${t1}$ ::  res, u, mask, n
+        integer :: zeros, bits_left, bits
+
+        n = scale
+        if(n <= 0_${k1}$) call error_stop("Error(unif_dist_rvs_1): Uniform"     &
+                         //" distribution scale parameter must be positive")
+        zeros = leadz(n)
+        bits = bit_size(n) - zeros
+        mask = shiftr(not(0_${k1}$), zeros)
+        L1 : do
+            u = dist_rand(n)
+            res = iand(u, mask)
+            if(res <= n) exit L1
+            bits_left = zeros
+            L2 : do
+                if(bits_left < bits) exit L2
+                u = shiftr(u, bits)
+                res = iand(u, mask)
+                if(res <= n) exit L1
+                bits_left = bits_left - bits
+            end do L2
+        end do L1
```

Comment:
```suggestion
        ${t1}$ ::  res, u, mask
        integer :: zeros, bits_left, bits

        if(scale <= 0_${k1}$) call error_stop("Error(unif_dist_rvs_1): Uniform"     &
                         //" distribution scale parameter must be positive")
        zeros = leadz(scale)
        bits = bit_size(scale) - zeros
        mask = shiftr(not(0_${k1}$), zeros)
        L1 : do
            u = dist_rand(scale)
            res = iand(u, mask)
            if(res <= scale) exit L1
            bits_left = zeros
            L2 : do
                if(bits_left < bits) exit L2
                u = shiftr(u, bits)
                res = iand(u, mask)
                if(res <= scale) exit L1
                bits_left = bits_left - bits
            end do L2
        end do L1
```
Just to avoid using `n`